spec/acceptance/ini_setting_spec.rb

```diff
@@ -101,34 +105,38 @@
   end
 
   context 'ensure parameter => absent for global' do
+    basedir = setup_test_directory
+
     before :all do
-      ini = <<-EOS
-        four = five
-        [one]
-        two = three
-        EOS
-      create_remote_file(default, File.join(tmpdir, 'ini_setting.ini'), ini)
+      ipp = <<-MANIFEST
+        file { '#{basedir}/ini_setting.ini':
+          content => "four = five\n [one]\n two = three",
+          force   => true,
+        }
+      MANIFEST
+
+      apply_manifest(ipp)
     end
+
     after :all do
-      shell("rm #{tmpdir}/ini_setting.ini", acceptable_exit_codes: [0, 1, 2])
+      run_shell("rm #{basedir}/ini_setting.ini", acceptable_exit_codes: [0, 1, 2])
```

Comment:
`run_shell` does not have `acceptable_exit_codes`. You should use `expect_failures: true` here.